### DISCUSSION TITLE
Aplication terminated due to LIBUSB_ERROR_TIMEOUT

### DISCUSSION THREAD

**Discussion by Vangasse** [bitcraze/discussions#1002] [Q&A] [answered]
I'm currently using Crazyflie 2.1 with a Crazyradio 2.0 in Ubuntu 20.04 with ROS Noetic and Crazyswarm lib.

As I send velocity commands every 100ms, the Crazyflie suddenly stops, and the error message is given as follows:

`terminate called after throwing an instance of 'std::runtime_error'
  what():  LIBUSB_ERROR_TIMEOUT`

Is anybody aware of this type of error? Thanks in advance.

### Edit:
It is very curious, as I experiment a fast technique with a computing time lesser than 30ms, calling `rospy.sleep()` for the rest of the loop time (100ms), everything works fine over the 60s experiment. Although, as I use a more complex technique with a computing time around 80ms, also sleeping through the rest of the 100ms, it flies for a few seconds until everything suddenly stops.

**Comment by knmcguire** [comment]
Hi! We need a bit more information. Are you running this on Ubuntu, windows or WSL in windows? Is the crazyflie pretty close to crazyradio or actually far away?

**Reply by Vangasse** [reply]
Hi, thanks for the reply.

I'm running on Ubuntu 20.04, the radio is about 1.5m up to 3m away from the crazyflie.

**Reply by knmcguire** [reply]
Hi! Sorry I complely missed this was with the crazyswarm2 package. It is probably better if you ask your question there or search if anybody has asked about it.

https://github.com/IMRCLab/crazyswarm2/discussions

**Comment by Vangasse** [comment, marked as answer]
I've updated the Crazyradio 2.0 firmware available at: https://github.com/bitcraze/crazyflie-link-cpp/files/12429056/crazyradio2-CRPA-emulation.zip

**Reply by knmcguire** [reply]
Did that help fix your issue?

**Reply by Vangasse** [reply]
Yes! I thought I had marked as answer, sorry!

**Reply by knmcguire** [reply]
ah great!

**Reply by knmcguire** [reply]
You don't have to close the discussion. or else it will be removed from the list. Just selecting the answer is fine :)
